src/utilities/core/Logger.hpp

```diff
@@ -115,15 +114,22 @@ class UTILITIES_API LoggerSingleton
   SinkSetType m_sinks;
 };
 
-#if _WIN32 || _MSC_VER
+class UTILITIES_API Logger
+{
+ public:
+  Logger() = delete;
 
-/// Explicitly instantiate and export LoggerSingleton Singleton template instance
-/// so that the same instance is shared between the DLL's that link to Utilities.dll
-UTILITIES_TEMPLATE_EXT template class UTILITIES_API openstudio::Singleton<LoggerSingleton>;
+  static LoggerImpl& instance() {
```

Comment:
I think I'd also like to follow Meyers' singleton pattern. Something feels a bit off here for me.

Reply:
I think we need to keep the name of the thing that has the `::instance` method as `Logger` otherwise we would break third party Measures. The question then becomes what do you call the thing that is returned by `Logger::instance`? I felt like LoggerSingleton was a really unfortunate name because it isn't a singleton (and the name at least confused me at first) so I renamed LoggerSingleton as LoggerImpl, which isn't great because it implies the Impl pattern that is used throughout OpenStudio, which it is not. I do think we could call it something else though, I just couldn't come up with a better name. I would be ok with renaming `LoggerImpl`as `Logger_Impl` and moving it under namesapce detail. Note that it is still a public thing, but then again so are all of our other Impls.

I don't think there's a particular reason why the static method definition is in the header and not the cpp. That is how the original code was and it didn't bother me enough to change.

The Meyers' singleton had one unfortunate side effect that was messing with us in this case, and that is it makes it easy to end up with multiple definitions. The singleton that I used here (phoenix singleton?) will yield a linker error if you don't link the one (and only) library where the singleton is defined. This helped me work through the issue. I also believe that we could use Meyers' singleton and move the defintion into the cpp and accomplish the same thing.

Reply:
You return a Logger.

```
class Logger {

public:
static Logger& instance() {
  static Logger logger;
  return logger;
}
  Logger(const Logger& other) = delete;
  Logger(Logger&& other) = delete;
  Logger& operator=(const Logger&) = delete;
  Logger& operator=(Logger&&) = delete;
private:
   Logger();

}
```

Reply:
I do follow the point about the Phoenix Singleton though.

Reply:
Yes, I do think that is enough. If the new test that I wrote is passing (on windows) then I am convinced.